### PR TITLE
[TTAHUB-2889] Save ActivityReportGoal FEI responses

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This change sets FEI root cause responses for multi recipient goals.

The two cases that have been updated are:
1. If a FEI goal is used on a multi recipient report were the goal already has FEI responses set.
2. Updates any AR responses when the FEI Root causes have been added to the goal AFTER the report has been created.

## How to test

- Review the code make sure you agree with the changes.
 
**Multi Recipient Report with FEI:**
1. Create a multi recipient report using an FEI goal.
2. Make sure the ActivityReportGoalFieldResponses have been created correctly.
3. Update the root causes make sure they are reflected on the ActivityReportGoalFieldResponses.
4. Approve the report and update the responses the old responses should remain.

**Multi Recipient Report with FEI missing responses:**
1. Create a multi recipient report using an FEI goal that has no responses set on either recipient.
2. Add FEI root causes to both recipient goals.
3. Verify that the ActivityReportGoalFieldResponses have been updated respectfully.
4. Approve the report.
5. Update the FEI root causes for both recipients make sure the old root causes remain.

**Single Recipient FEI root cause:** 
Make sure we can still create a single recipient report and the root causes still follow last save wins until the report is approved.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2889

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: src/models/hooks/goalFieldResponse.js [hunk 1]
```diff
@@ -70,6 +72,32 @@ const syncActivityReportGoalFieldResponses = async (sequelize, instance, _option
       },
     );
 
+    // Get all the ActivityReportGoals that are missing a response.
+    const missingResponses = await sequelize.models.ActivityReportGoal.findAll({
```
Comment (AdamAdHocTeam) [inline src/models/hooks/goalFieldResponse.js:76]: This code is responsible for creating any missing activity report goal field responses that might have existed before the prompts were added.

File: src/services/reportCache.js [hunk 2]
```diff
@@ -450,22 +453,40 @@ const cacheGoalMetadata = async (
     Goal.update({ onAR: true }, { where: { id: goal.id }, individualHooks: true }),
   ];
 
-  if (!isMultiRecipientReport && prompts && prompts.length) {
-    finalPromises.push(
-      cachePrompts(goal.id, arg.id, prompts),
-    );
-  }
-
   if (isMultiRecipientReport) {
```
Comment (AdamAdHocTeam) [inline src/services/reportCache.js:456]: This code is responsible for creating the activity report goal field responses at report creation time if the fei root causes already exist on the goal.

**Review by thewatermethod — APPROVED** [review]

File: src/services/reportCache.test.js [hunk 3]
```diff
@@ -462,7 +601,7 @@ describe('cacheObjectiveMetadata', () => {
     await Promise.all(roles.map(async (role) => role.destroy()));
     await Promise.all(files.map(async (file) => file.destroy()));
     await User.destroy({ where: { id: user.id } });
-    await db.sequelize.close();
+    db.sequelize.close();
```
[the remark this answers is not in the document]
Comment (AdamAdHocTeam) [reply src/services/reportCache.test.js]: This has been added back.

File: src/models/hooks/goalFieldResponse.js [hunk 4]
```diff
@@ -56,20 +56,42 @@ const syncActivityReportGoalFieldResponses = async (sequelize, instance, _option
...
+        returning: true,
       },
     );
 
+    // Get all the ActivityReportGoal ids that were updated.
+    const updatedResponseIds = [
+      ...new Set(updatedResponses[1].map((item) => item.activityReportGoalId)),
+    ];
+
+    // If the activityReportGoal exists but wasn't updated we know its missing a response.
```
Comment (GarrettEHill) [inline src/models/hooks/goalFieldResponse.js:81]: Is this always a true statement?

Comment (AdamAdHocTeam) [reply src/models/hooks/goalFieldResponse.js:81]: Ok I will write a test for this case to ensure if its already the value in the update it still comes back in the updated list.

Comment (AdamAdHocTeam) [reply src/models/hooks/goalFieldResponse.js:81]: I have added a test for this case.

**Review by GarrettEHill — APPROVED** [review]

**Review by kryswisnaskas — APPROVED** [review]
Looks good! 👍 

I was able to verify that the ActivityReportGoalFieldResponses records are created in the database.

**Pull request merged**